server/server/data_loader/arangoload.py

```diff
@@ -183,6 +194,100 @@ def process_division_files(
     docs.append({'uid': 'orphan', '_key': 'orphan'})
 
 
+def process_names_files(
+        names_files: List[Path],
+        root_languages: Dict[str, str],
+        super_extra_info:  Dict[str, Dict[str, str]],
+        text_extra_info:  Dict[str, Dict[str, str]]
+) -> List[dict]:
+    """
+    Method for processing name files from sc-data/structure/name
+
+    Args:
+        names_files - list of name Path objects to files from name folder
+        root_languages - parsed data from language.json
+        super_extra_info - parsed data from super_extra_info.json
+        text_extra_info - parsed data from text_extra_info.json
+
+    Returns:
+        list of processed data
+    """
+    docs = []
+    names_files.sort(key=lambda path: len(path.parts))
+    for name_file in names_files:
+        entries: Dict[str, str] = json_load(name_file)
+        for uid, name in entries.items():
+            if not name or type(name) != str:
```

Comment:
how can `type(name)` not be a string? is that a data problem?

Reply:
unfortunately yes
for example this file: 
https://github.com/suttacentral/sc-data/blob/2606d251c73ec72e3212ed3ea92dbff4c41f8f45/structure/name/root/abhidhamma/pr-name.json#L2

but then we could found such uid here:
https://github.com/suttacentral/sc-data/blob/2606d251c73ec72e3212ed3ea92dbff4c41f8f45/structure/name/root/super-name.json#L201

that's why we need such "check"
maybe you have a better idea? :thinking: